shimmingtoolbox/cli/create_coil_profiles.py

```diff
@@ -291,6 +296,147 @@ def create_coil_profiles_cli(fname_json, path_relative, autoscale, unwrapper, th
+@click.option('--coil_name', 'coil_name', required=False, type=click.STRING, default="new",
+              help="Name of the coil. If not provided, \"new\" will be used.")
+@click.option('--min', 'min_current', required=False, type=click.FLOAT, default=-1,
+              help="The minimum current in amps going through each channel. Defaulted to -1 A.")
+@click.option('--max', 'max_current', required=False, type=click.FLOAT, default=1,
+              help="The maximum current in amps going through each channel. Defaulted to 1 A.")
+@click.option('--max_sum', 'max_current_sum', required=False, type=click.FLOAT, default=None,
+              help="The maximum sum of currents in amps going through all loops. Defaulted to the number of channel.")
+@click.option('-o', '--output', 'fname_output', type=click.Path(), required=False,
+              default=os.path.join(os.path.curdir, 'coil_profiles.nii.gz'),
+              help="Output filename of the coil profiles NIfTI file. Supported types : '.nii', '.nii.gz'")
+@click.option('-v', '--verbose', type=click.Choice(['info', 'debug']), default='info', help="Be more verbose")
+def from_cad(fname_txt, fname_fmap, offset, dims_to_flip, software, coil_name, min_current, max_current, max_current_sum, fname_output, verbose):
+    """ Testing change Create \u0394B\u2080 coil profiles from CAD wire geometries."""
+    # Assert inputs
+    assert min_current < max_current, f"Minimum current should be smaller than maximum current ({min_current} >= {max_current})"
+    # Set logger level
+    set_all_loggers(verbose)
+
+    # create the output folder
+    create_output_dir(fname_output)
+
+    # Set variables
+    GAMMA = 42.576E6 # in Hz/Tesla
+    nif = nib.load(fname_fmap)
+    pmcn = CAD_to_pumcin(fname_txt, list(dims_to_flip), software)
+    pmcn[:, 1:4] += np.array(list(offset))
+    Wires = get_wire_pattern(pmcn)
+    transform = nif.affine[:-1, :]
+    nb_channels = len(Wires)
+
+    # Map the position (in mm) of all pixel in the FOV
+    FOV_shape = nif.header.get_data_shape()
+    xx = np.arange(FOV_shape[0])
+    yy = np.arange(FOV_shape[1])
+    zz = np.arange(FOV_shape[2])
+    X, Y, Z = np.meshgrid(xx, yy, zz, indexing='ij')
+    voxel_coords = np.array([X.ravel(order='F'), Y.ravel(order='F'), Z.ravel(order='F')])
+    voxel_coords = np.vstack((voxel_coords, np.ones(voxel_coords.shape[1])))
+    world_coords = transform @ voxel_coords
+    gridSize = X.shape
+
+    # Generate the coil profiles
+    coil_profiles = np.zeros((gridSize[0], gridSize[1], gridSize[2], len(Wires)))
+    for iCh in range(len(Wires)):
+        coil_profiles[:, :, :, iCh] = generate_coil_bfield(Wires[iCh], world_coords.T, gridSize)
+    coil_profiles *= GAMMA
+
+    # Save the coil profiles
+    affine = nif.affine
+    header = nif.header
+    nii = nib.Nifti1Image(coil_profiles, affine=affine, header=header)
+    nib.save(nii, os.path.join(fname_output, coil_name + "_coil_profiles.nii.gz"))
```

Comment:
The default value of fname_output is a filename which will lead to unexpected behaviour if you don't provide the `-o` option